src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -217,12 +225,25 @@ private MetricValue GetMostRecentMetricValue(string metricName, TimeSeriesElemen
             return metricQuery;
         }
 
-        private IAzure CreateAzureClient(AzureEnvironment azureCloud, string tenantId, string subscriptionId, string applicationId, string applicationSecret, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory, MetricSinkWriter metricSinkWriter, IRuntimeMetricsCollector metricsCollector)
+        private IAzure CreateAzureClient(AzureEnvironment azureCloud, string tenantId, string subscriptionId, AuthenticationMode authenticationMode, string managedIdentityId, string applicationId, string applicationSecret, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory, MetricSinkWriter metricSinkWriter, IRuntimeMetricsCollector metricsCollector)
         {
-            var credentials = _azureCredentialsFactory.FromServicePrincipal(applicationId, applicationSecret, tenantId, azureCloud);
+            AzureCredentials credentials;
+
+            if (authenticationMode == AuthenticationMode.ManagedIdentity && !string.IsNullOrEmpty(managedIdentityId))
+            {
+                credentials = _azureCredentialsFactory.FromUserAssigedManagedServiceIdentity(managedIdentityId, MSIResourceType.VirtualMachine, azureCloud, tenantId);
+            }
+            else if (authenticationMode == AuthenticationMode.ManagedIdentity)
+            {
+                credentials = _azureCredentialsFactory.FromSystemAssignedManagedServiceIdentity(MSIResourceType.VirtualMachine, azureCloud, tenantId);
+            }
+            else
+            {
+                credentials = _azureCredentialsFactory.FromServicePrincipal(applicationId, applicationSecret, tenantId, azureCloud);
+            }
```

Comment:
I would move this to a dedicated method
```suggestion
            AzureCredentials credentials = CreateAzureAuthentication(azureCloud, authenticationMode, tenantId, managedIdentityId);
```

Reply:
As you want !